service/src/main/java/uk/gov/hmcts/reform/fpl/service/UploadDocumentService.java

```diff
@@ -44,4 +44,13 @@ public Document uploadPDF(byte[] pdf, String fileName) {
 
         return document;
     }
+
+    public String getUploadedDocumentUserInfo() {
+        String userToken = idamClient.getAccessToken(userConfig.getUserName(), userConfig.getPassword());
+        UserDetails userDetails = idamClient.getUserByUserId(userToken, requestData.userId());
+
+        boolean isHMCTSUser = userDetails.getRoles().stream().anyMatch(UserRole::isHMCTSUser);
+
+        return isHMCTSUser ? "HMCTS": userDetails.getEmail();
+    }
```

Comment:
Can this go here or any other existing service? I think i probably prefer a new service. Any suggestion please?